src/amo/utils/addonCache.ts

```diff
@@ -0,0 +1,64 @@
+import * as fs from "fs";
+import * as path from "path";
+
+/***
+ * The cache folder is stored at
+ * ../Code/User/globalStorage/undefined_publisher.assay/.cache
+ * Upon initialization, the extension folder, undefined_publisher.assay,
+ * may not exist. This function creates that folder and the cache folder
+ * if they do not exist.
+ *
+ * The folder undefined_publisher.assay will continue to exist after this.
+ */
+export async function addonInfoToCache(
+  storagePath: string,
+  addonGUID: string,
+  key: string,
+  value: string
+) {
+  const cacheFolderPath = path.join(storagePath, ".cache");
+  const cacheFilePath = path.join(cacheFolderPath, `${addonGUID}.json`);
+
+  if (!fs.existsSync(storagePath)) {
+    fs.mkdirSync(storagePath);
+  }
+
+  if (!fs.existsSync(cacheFolderPath)) {
+    fs.mkdirSync(cacheFolderPath);
+  }
+
+  if (!fs.existsSync(cacheFilePath)) {
+    fs.writeFileSync(cacheFilePath, JSON.stringify({ [key]: value }));
+  } else {
+    const cacheFile = fs.readFileSync(cacheFilePath, "utf-8");
+    const cacheFileJSON = JSON.parse(cacheFile);
+    cacheFileJSON[key] = value;
+    fs.writeFileSync(cacheFilePath, JSON.stringify(cacheFileJSON));
+  }
```

Comment:
There's a bit of duplication of logic here, so how about assuming the file exists and catching the exception instead.  something like (untested code ahead):
``` javascript
let cacheFileJSON = {};
try {
    const cacheFile = fs.readFileSync(cacheFilePath, "utf-8");
    cacheFileJSON = JSON.parse(cacheFile);
} catch {}
cacheFileJSON[key] = value;
fs.writeFileSync(cacheFilePath, JSON.stringify(cacheFileJSON));
```